deeplearning4j-nn/src/main/java/org/deeplearning4j/optimize/listeners/TimeIterationListener.java

```diff
@@ -0,0 +1,74 @@
+/*-
+ *
+ *  * Copyright 2015 Skymind,Inc.
+ *  *
+ *  *    Licensed under the Apache License, Version 2.0 (the "License");
+ *  *    you may not use this file except in compliance with the License.
+ *  *    You may obtain a copy of the License at
+ *  *
+ *  *        http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  *    Unless required by applicable law or agreed to in writing, software
+ *  *    distributed under the License is distributed on an "AS IS" BASIS,
+ *  *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *    See the License for the specific language governing permissions and
+ *  *    limitations under the License.
+ *
+ */
+
+package org.deeplearning4j.optimize.listeners;
+
+import org.deeplearning4j.nn.api.Model;
+import org.deeplearning4j.optimize.api.IterationListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+
+/**
+ * Time Iteration Listener.
+ * This listener displays into INFO logs the remaining time in minutes and the date of the end of the process. 
+ */
+public class TimeIterationListener implements IterationListener {
+
+    private static final long serialVersionUID = 1L;
+    private boolean invoked;
+    private long start;
+    private int iterationCount;
+    private static final Logger log = LoggerFactory.getLogger(TimeIterationListener.class );
+    
+    /**
+     * Constructor
+     * @param iterationCount The global number of iteration of the process 
+     */
+    public TimeIterationListener( int iterationCount )
+    {
+        this.iterationCount = iterationCount;
+        start = System.currentTimeMillis();
+    }
+    
+    @Override
+    public boolean invoked()
+    {
+        return invoked;
+    }
+
+    @Override
+    public void invoke()
+    {
+        invoked = true;
+    }
+
+    @Override
+    public void iterationDone( Model model, int iteration)
+    {
+        int currentIteration = iteration + 1;
```

Comment:
This thing. In ParallelWrapper environment, this Listener instance will be shared between multiple workers, each of them will have own iteration number. So we just need to use AtomicLong counter and increment it, so this Listener will be also perfectly compatible with PW